build-macos

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+set -e
+while [ $# -gt 0 ]; do
+  PKG_SCRIPTNAME=$1
+  shift
+
+  PKG_ARCH=x86_64
+  export CPP=cpp
+  export AR=ar
+  export AS=as
+  export NM=nm
+  export CC=clang
+  export CXX=clang++
+  export LD=ld
+  export RANLIB=ranlib
+  export STRIP=strip
+  export CFLAGS="-O2 ${CFLAGS}"
+  export CXXFLAGS="-O2 ${CXXFLAGS}"
+  export PKG_OS=macos
+  export PKG_ARCH
```

Comment:
I have a sense that most of these variables are unnecessary, because they're actually setting the default value and, as such, they can be omitted. At the same time, one critical bit regarding building on macOS is that I am not sure about how to _guarantee_ that libs that we're building are picked up, instead of system wide ones. It may be that doing that works out of the box, or that it will not and we'll need to do some work. (Saying this just as a reminder to self that this is important). Finally: do you also plan on shipping `i386` binaries for macOS?

Reply:
It would be nice to have `i386` binaries, but it's not strictly needed.

Reply:
Good